compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract/Exports.hs

```diff
@@ -0,0 +1,141 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- | This module processes export lists and then answers the question
+-- "is this *thing* exported?"
+module DA.Daml.Doc.Extract.Exports
+    ( ExportSet
+    , extractExports
+    , exportsType
+    , exportsConstr
+    , exportsFunction
+    , exportsField
+    ) where
+
+import DA.Daml.Doc.Types as DD
+
+import "ghc-lib" GHC
+import "ghc-lib-parser" RdrName
+import "ghc-lib-parser" OccName
+import "ghc-lib-parser" FieldLabel
+import "ghc-lib-parser" FastString
+
+import qualified Data.Set as Set
+import qualified Data.Text as T
+
+-- | Set of module exports.
+--
+-- Unlike Haddock, we don't ask the export list to dictate the order
```

Comment:
Nothing for this PR but I kind of like Haddocks behavior here.

Reply:
Yep. It's an option for damldocs, but it implies changing a bunch of things.